### PR TITLE
[Refactor:Forum] move inline JS

### PR DESCRIPTION
**Pull request by zeez2030**
### Please check if the PR fulfills these requirements:

*  The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?

Closes #5650 
### What is the new behavior?
A new separate file stat_page.js is added that includes all the inline script that was found inside StatPage.twig

### CODE REVIEW

**Comment by zeez2030** [comment]
@marwanatef2

**Comment by zeez2030** [comment]
> Hii @zeez2030 thanks for your contribution!
> Please rename the `stat_page.js` to `stat-page.js` to be consistent with rest of the JS file naming

@mukul-kmr-jha  done

File: site/public/js/stat-page.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+$('td').click(function () {
```
Comment (MasterOdin) [inline site/public/js/stat-page.js:1]: You will need to add `/* global sortTable, escapeSpecialChars, buildCourseUrl */` to get this file to pass the linter as those  functions are defined elsewhere.

Comment (zeez2030) [reply site/public/js/stat-page.js:1]: just to understand you clearly, do you mean that I should implement them from scratch or import them from a file somewhere

**Review by MasterOdin — APPROVED** [review]

**Comment by MasterOdin** [comment]
Thanks for this. Sorry it took so long to merge.

**Pull request merged**